astropy/table/table.py

```diff
@@ -3379,6 +3379,79 @@ def _replace_cols(self, columns):
 
         self.columns = columns
 
+    def setdefault(self, name, default):
+        """Ensure a column named ``name`` exists.
+
+        If ``name`` is already present then ``default`` is ignored.
+        Otherwise ``default`` can be any data object which is acceptable as
+        a `~astropy.table.Table` column object or can be converted.  This
+        includes mixin columns and scalar or length=1 objects which get
+        broadcast to match the table length.
+
+        Parameters
+        ----------
+        name : str
+            Name of the column.
+        default : object
+            Data object for the new column
+
+        Returns
+        -------
+        Column, MaskedColumn or mixin-column type
+            The column with the name ``col`` if it is present already,
```

Comment:
It took me a while to figure out a good name for the argument. I finally chose `name` because that is what e.g. `Table.add_column()` uses, but I failed to make the change in this line.